### PR TITLE
Add Software Composition Analysis Scan

### PR DESCRIPTION
**Pull request by imda-benedictlee**
## Description

Add Blackduck Scanning On Pull Request & Work Dispatch

## Motivation and Context

To enable Vulnerabilities & License Scans On Pull Request & Work Dispatch

## Type of Change

- ci: Changes to the CI/CD configuration or scripts

## How to Test

Run Blackduck Scanning Workflow On Pull Request & Work Dispatch

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [ ]  I have added or updated the necessary documentation (README, API docs, etc.).
- [ ]  I have added appropriate unit tests or functional tests for the changes made.
- [x]  I have followed the project's coding conventions and style guidelines.
- [x]  I have rebased my branch onto the latest commit of the main branch.
- [x]  I have squashed or reorganized my commits into logical units.
- [ ]  I have added any necessary dependencies or packages to the project's build configuration.
- [ ]  I have performed a self-review of my own code.
- [x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Additional Notes

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

File: .github/workflows/sca_scan.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    - name: Set Branch Variable (Workflow_Dispatch)
+      if: github.event_name == 'workflow_dispatch'
+      run: |
+          echo "BRANCH=${{ inputs.branch_to_test }}" >> "$GITHUB_ENV"
+
+    - name: Install & Scan MOONSHOT
+      run: |
+          echo $BRANCH
+          cd moonshot
+          npm install
```
Comment (imda-normanchia) [inline .github/workflows/sca_scan.yaml:60]: sorry but can check why you need to run npm install for moonshot?

Comment (imda-benedictlee) [reply .github/workflows/sca_scan.yaml:60]: oh...my bad..will update it accordingly. thanks for pointing it out!

Comment (imda-benedictlee) [reply .github/workflows/sca_scan.yaml]: Fix Python Installation

**Review by imda-kwokwk — APPROVED** [review]
ok

**Review by miyamaya9 — APPROVED** [review]

**Pull request merged**